### PR TITLE
Fix incorrect msg->len calculation during redirect preparation.

### PR DESCRIPTION
**Pull request by EvgeniiMekhanik**
msg len calculation was totally wrong: we should adjust status line len instead of length of S_301, because we can use different status codes, also we should adjust S_F_SET_COOKIE length.

### CODE REVIEW

File: fw/http.c [hunk 1]
```diff
@@ -808,16 +809,18 @@ do { 								\
...
 		body_val = p;
 		body_len = tfw_str_to_cstr(body, body_val, remaining);
 	}
 
+	status_line_len = strlen(status_line);
 	{
 		TfwStr msg = {
 			.chunks = (TfwStr []){
 				{ .data = status_line,
-				  .len = strlen(status_line) },
```
[the remark this answers is not in the document]
Comment (EvgeniiMekhanik) [reply fw/http.c:820]: Yes fixed

File: fw/http.c [hunk 2]
```diff
@@ -815,7 +813,7 @@ do { 								\
 		body_len = tfw_str_to_cstr(body, body_val, remaining);
 	}
 
-	status_line_len = strlen(status_line);
+	status_line_len = sizeof(status_line);
```
Comment (const-t) [inline fw/http.c:816]: `sizeof` return size of pointer in this case.

**Review by const-t — CHANGES_REQUESTED** [review]

File: fw/http.c (hunk 2 above)
Comment (EvgeniiMekhanik) [reply fw/http.c:816]: Yes I make this change automatically according our discussion(

**Review by krizhanovsky — APPROVED** [review]
LGTM

**Review by const-t — APPROVED** [review]

**Pull request merged**